lib/openstack_query/query_blocks/query_output.py

```diff
@@ -153,9 +175,51 @@ def generate_output(
         :param openstack_resources: List of openstack objects to obtain properties from - e.g. [Server1, Server2]
         :return: List containing dictionaries of the requested properties obtained from the items
         """
-        return [self._parse_property(item) for item in openstack_resources]
+        output = []
+        for item in openstack_resources:
+            prop_list = self._parse_properties(item)
+            prop_list.update(self._parse_forwarded_outputs(item))
+            output.append(prop_list)
+        return output
 
-    def _parse_property(
+    def _parse_forwarded_outputs(
+        self, openstack_resource: OpenstackResourceObj
+    ) -> Dict[str, str]:
+        """
+        Generates a dictionary of forwarded outputs for the item given
+        :param openstack_resource: openstack resource item to parse forwarded outputs for
+        """
+        forwarded_output_dict = {}
+        if self.forwarded_outputs:
```

Comment:
This should always be a dict, even if it's empty. Lets drop this `if` and let the for handle the case where there's 0 elements